pkg/provider/pulumi/generator.go

```diff
@@ -130,9 +151,85 @@ func (a *pulumiDeployment) Delete(name string) error {
 	if err != nil {
 		return err
 	}
-	res, err := s.Destroy(context.Background())
+
+	var opts []optdestroy.Option
+	upChannel := make(chan events.EngineEvent)
+	opts = append(opts, optdestroy.EventStreams(upChannel))
+	go collectEvents(upChannel)
+
+	if output.VerboseLevel >= 2 {
+		opts = append(opts, optdestroy.ProgressStreams(os.Stdout))
+	}
+	if output.VerboseLevel > 2 {
+		var loglevel uint = uint(output.VerboseLevel)
+		opts = append(opts, optdestroy.DebugLogging(debug.LoggingOptions{
+			LogLevel:      &loglevel,
+			LogToStdErr:   true,
+			FlowToPlugins: true,
+		}))
+	}
+
+	res, err := s.Destroy(context.Background(), opts...)
 	if err != nil {
 		return errors.WithMessage(err, res.Summary.Message)
 	}
 	return nil
 }
+
+func stepEventToString(eType string, evt *apitype.StepEventMetadata) (string, string) {
+	action := string(evt.Op)
+	if evt.Op == apitype.OpSame {
+		action = "unchanged"
+	}
+
+	urnSplit := strings.Split(evt.URN, "::")
+	name := urnSplit[len(urnSplit)-1]
+
+	typeSplit := strings.Split(evt.Type, ":")
+	rType := typeSplit[len(typeSplit)-1]
+
+	return name, fmt.Sprintf("%s %s:%s", action, rType, name)
+}
+
+func collectEvents(eventChannel <-chan events.EngineEvent) {
+	lastCreating := ""
+	creatingMsg := ""
+	for {
+		var event events.EngineEvent
+		var ok bool
+
+		event, ok = <-eventChannel
+		if !ok {
+			return
+		}
+		// this is so the message is to the right of the "Deploying ..."
+		spacer := "                     "
+
+		if event.ResourcePreEvent != nil && event.ResourcePreEvent.Metadata.Op != apitype.OpSame {
+			lastCreating, creatingMsg = stepEventToString("ResourcePreEvent", &event.ResourcePreEvent.Metadata)
+			pterm.Info.Print(spacer + creatingMsg)
```

Comment:
We could avoid the output interleaving (spinner being interrupted by output) by using pterm to print our debug output as well e.g. pter.Debug.Print(). This output style co-operates with the spinner on stdout rather than competing.